tests/components/hue_ble/test_config_flow.py

```diff
@@ -0,0 +1,240 @@
+            1,
+            True,
+            False,
+            True,
+            (True, []),
+            pytest.raises(InvalidAuth),
+        ),
+        (
+            generate_ble_device(TEST_DEVICE_NAME, TEST_DEVICE_MAC),
+            1,
+            True,
+            None,
+            True,
+            (True, []),
+            does_not_raise(),
+        ),
+        (
+            generate_ble_device(TEST_DEVICE_NAME, TEST_DEVICE_MAC),
+            1,
+            True,
+            True,
+            False,
+            (True, []),
+            pytest.raises(CannotConnect),
+        ),
+        (
+            generate_ble_device(TEST_DEVICE_NAME, TEST_DEVICE_MAC),
+            1,
+            True,
+            True,
+            True,
+            (True, ["Error :P"]),
+            pytest.raises(CannotConnect),
+        ),
+    ],
+    ids=[
+        "good_data",
+        "no_scanners",
+        "not_found",
+        "invalid_auth",
+        "unknown_auth_status",
+        "cannot_connect",
+        "failed_poll",
+    ],
+)
+async def test_validation(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    return_device_result,
+    scanner_count_result,
+    connect_result,
+    authenticated_result,
+    connected_result,
+    poll_state_result,
+    expected_result,
+) -> None:
+    """Test input validation."""
+    with (
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_ble_device_from_address",
+            return_value=return_device_result,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.async_scanner_count",
+            return_value=scanner_count_result,
+        ),
+        patch(
+            "homeassistant.components.hue_ble.config_flow.HueBleLight",
+            autospec=True,
+            authenticated=authenticated_result,
+            connected=connected_result,
+        ) as mock_obj,
+    ):
+        client = mock_obj.return_value
+        client.connect.return_value = connect_result
+        client.authenticated = authenticated_result
+        client.connected = connected_result
+        client.poll_state.return_value = poll_state_result
+
+        with expected_result:
+            await validate_input(hass, TEST_DEVICE_MAC)
+
+
+async def test_bluetooth_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test bluetooth discovery form."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_BLUETOOTH},
+        data=HUE_BLE_SERVICE_INFO,
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "confirm"
+    assert result["description_placeholders"] == {
+        CONF_NAME: TEST_DEVICE_NAME,
+        CONF_MAC: TEST_DEVICE_MAC,
+        "url_pairing_mode": URL_PAIRING_MODE,
+    }
+
+    with patch(
+        "homeassistant.components.hue_ble.config_flow.validate_input",
+        return_value=True,
+    ):
```

Comment:
We should not patch internals, instead mock out the library and make sure it responds like we expect